### PR TITLE
[GEOS-10384] Change GetMap to URIKvpParser

### PR DESCRIPTION
**Pull request by jodygarnett**
[GEOS-10384](https://osgeo-org.atlassian.net/browse/GEOS-10384)

By using URI we can avoid creating URL until needed for connection.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by jodygarnett** [comment]
Okay, if I deprecate a class can I still keep a test case for it?
```
Warning: 525 [WARNING] /home/runner/work/geoserver/geoserver/src/ows/src/test/java/org/geoserver/ows/kvp/URLKvpParserTest.java:[20,22] org.geoserver.ows.kvp.URLKvpParser in org.geoserver.ows.kvp has been deprecated
```

**Comment by aaime** [comment]
You can, by suppress the warning (as I see you did).

**Review by aaime — COMMENTED** [review]
Generally looks good, noted that there is still stuff talking about URLs when they are in fact URI

File: src/wms/src/main/java/org/geoserver/wms/GetMapRequest.java [hunk 1]
```diff
@@ -157,7 +158,7 @@ public URL getSld() {
      *
      * <p>This parameter is used to point to a remote style via url.
```
[the remark this answers is not in the document]
Comment (jodygarnett) [reply src/wms/src/main/java/org/geoserver/wms/GetMapRequest.java:159]: Yes I will fix, thanks!

**Comment by jodygarnett** [comment]
Thanks for the review @aaime, feedback addressed, will merge and make a new PR to back-port.

**Pull request merged**